servicenow/getlogs_servicenow-3.8.py

```diff
@@ -0,0 +1,1644 @@
+#!/usr/bin/env python
+import configparser
+import json
+import logging
+import os
+import regex
+import socket
+import sys
+import time
+import pytz
+from optparse import OptionParser
+from multiprocessing import Process
+from datetime import datetime
+import dateutil
+import urllib.parse
+import http.client
+import requests
+import statistics
+import subprocess
+import shlex
+import ifobfuscate
+import re
+import sqlite3
+
+'''
+This script gathers data to send to Insightfinder
+'''
+
+location_map = {}
+RESULT_KEY = "result"
+LOCATION_KEY = "location"
+LINK_KEY = "link"
+VALUE_KEY = "value"
+NAME_KEY = "name"
+
+
+def start_data_processing(thread_number):
+    # set sysparm limi/offset
+    passthru = {'sysparm_limit': 100,
+                'sysparm_exclude_reference_link': 'true',
+                'sysparm_display_value': 'true',
+                'sysparm_offset': agent_config_vars['state']['sysparm_offset'],
+                'sysparm_query': ''}
+
+    ## add timestamp query
+    # get utc earliest datetime
+    utc_earliest_epoch = time.time()
+    # get localized earliest datetime
+    local_earliest_datetime = datetime.fromtimestamp(utc_earliest_epoch)
+    # convert earliest datetime to the data timezone
+    data_earliest_datetime = local_earliest_datetime.astimezone(
+        agent_config_vars['timezone'])
+    # do not apply timezone conversion later
+    # agent_config_vars['timezone'] = pytz.utc
+    # convert to string for Glide
+    earliest_date_and_time = data_earliest_datetime.strftime(
+        agent_config_vars['timestamp_format'][0]).split(' ')
```

Comment:
The condition for this code is if the historical flag is not set and there is no status file with a cron start time.  We should move it to the related section of code (line 83).